syntax = "proto2";

// make new paste
message newPasteArgs {
    required int64 user_id = 1;
    required string text = 2;
    optional string title = 3;
    optional string password = 4;
};

message newPasteResponce {
    required string public_key = 1;
}

// get paste info
message getPasteInfoArgs {
    required string public_key = 1;
}

message getPasteInfoResponce {
    required string author = 1;
    required string password = 2;
    required string title = 3;
    required string created_at = 4;
}

// get paste text
message getPasteTextArgs {
    required string public_key = 1;
}

message getPasteTextResponce {
    required string paste_text = 1;
}

// delete paste
message delPasteArgs {
    required string public_key = 1;
}

message delPasteResponce {
    required bool is_success = 1;
}

// update paste info
message updatePasteArgs {
    required string public_key = 1;
    optional string new_password = 2;
    optional string new_title = 3;
}

message updatePasteResponce {
    required bool is_success = 1;
}

// get last n pastes
message getLastePastesArgs {
    required int64 user_id = 1;
    required int32 count = 2;
}

message getLastePastesResponce {
    required string last_pastes = 1;
}

service pastebinApi {
    rpc makeNewPaste(newPasteArgs) returns(newPasteResponce) {}
    rpc getPasteInfo(getPasteInfoArgs) returns(getPasteInfoResponce) {}
    rpc getPasteText(getPasteTextArgs) returns(getPasteTextResponce) {}
    rpc deletePaste(delPasteArgs) returns(delPasteResponce) {}
    rpc updatePaste(updatePasteArgs) returns(updatePasteResponce) {}
    rpc getLastPastes(getLastePastesArgs) returns(getLastePastesResponce) {}
}


